src/main/scala/ot/geckopipe/index/V2DIndex.scala

```diff
@@ -105,8 +105,44 @@ object V2DIndex extends LazyLogging {
         .drop("chr_id", "position", "ref_allele", "alt_allele"))
   }
 
-  def buildStudiesIndex(path: String)(implicit ss: SparkSession): DataFrame =
-    ss.read.parquet(path).orderBy(col("study_id").asc)
+  def buildStudiesIndex(path: String, efos: String)(implicit ss: SparkSession): DataFrame = {
+    import ss.implicits._
+
+    val efoColumns = List(
+      "study_id",
+      "trait_efos",
+      "trait_category"
+    )
+
+    val efoDF = ss.read
+      .parquet(efos)
+      .select(efoColumns.head, efoColumns.tail: _*)
+      .orderBy(efoColumns.head)
+
+    val pattern = """^([a-zA-Z]+)(.*)"""
+    val studies = ss.read
+      .parquet(path)
+      .withColumn("pmid", when(length($"pmid") > 0, $"pmid"))
+      .withColumn("pub_date", when(length($"pub_date") > 0, $"pub_date"))
```

Comment:
What are these `when(length(<col>) > 0` used for? If I'm reading it correctly if the length is greater than 0 it returns the value that was there, otherwise it returns null. Couldn't you just have a 'select ... as ...' and not use `withColumn`?

Reply:
@JarrodBaker, this is a defence measure as the unstructured data will bring things like `[""]` or `""` so we are making sure this won't pass into the output data we generate.